csrc/scheduler/tools/loop_domain_scheduler.cpp

```diff
@@ -385,5 +385,102 @@ void scheduleLoopDomainsLike(
+
+  const ValGroups input_groups = exact_graph.toGroups(transform->inputs());
+  const ValGroups output_groups = exact_graph.toGroups(transform->outputs());
+
+  for (auto tv : tvs) {
+    const ValGroups loop_groups = exact_graph.toGroups(tv->getLoopDomain());
+
+    // Check if either the inputs or the outputs are mapped with the
+    // loop domain.
+
+    std::vector<IterDomain*> input_ids;
+    input_ids.reserve(transform->inputs().size());
+    for (const auto& input_g : input_groups) {
+      for (const auto loop_id : tv->getLoopDomain()) {
+        if (input_g->has(loop_id)) {
+          input_ids.push_back(loop_id);
+        }
+      }
+    }
+
+    std::vector<IterDomain*> output_ids;
+    output_ids.reserve(transform->outputs().size());
+    for (const auto& output_g : output_groups) {
+      for (const auto loop_id : tv->getLoopDomain()) {
+        if (output_g->has(loop_id)) {
+          output_ids.push_back(loop_id);
+        }
+      }
+    }
+
+    Direction replay_dir = Direction::Undefined;
+
+    // It should be either: all of the inputs found and none of the
+    // outputs found, or none of the inputs found and all of the
+    // outputs found.
+    if (input_ids.size() == transform->inputs().size()) {
+      NVF_ERROR(output_ids.empty());
+      replay_dir = Direction::Forward;
+    } else if (output_ids.size() == transform->outputs().size()) {
+      NVF_ERROR(input_ids.empty());
+      replay_dir = Direction::Backward;
+    } else {
+      // Replay not possible since none of inputs nor outputs are connected with
+      // the transform
+      continue;
+    }
+
+    const auto& existing_ids =
+        replay_dir == Direction::Forward ? input_ids : output_ids;
+
+    // Clone inputs or outputs
+    auto& new_ids = replay_dir == Direction::Forward ? output_ids : input_ids;
+    const auto& ref_of_ids_to_generate = replay_dir == Direction::Forward
+        ? transform->outputs()
+        : transform->inputs();
+
+    for (const auto& ref_id : ref_of_ids_to_generate) {
+      auto clone = ref_id->as<IterDomain>()->cloneWithoutRFactor();
+      new_ids.push_back(clone);
+    }
+
+    // In the case of replaying the transform expr backward,
+    // the definition of the output IDs will be set to the newly
+    // created expr. This is only allowed when the output IDs have no
+    // definition yet.
+    LoopDomainSchedulerReplayTransform::replayAs(
+        input_ids, output_ids, transform);
+
+    // Replace the inputs of the transform with the outputs
+    auto new_loop_domain = tv->getLoopDomain();
+    auto outermost_pos = (int64_t)tv->getLoopDomain().size();
+    for (const auto& existing_id : existing_ids) {
+      auto it = std::find(
+          new_loop_domain.begin(), new_loop_domain.end(), existing_id);
+      NVF_ERROR(it != new_loop_domain.end());
+      auto pos = (int64_t)std::distance(new_loop_domain.begin(), it);
+      outermost_pos = std::min(outermost_pos, pos);
```

Comment:
I think this means that if we have something like
```
i2 = merge(i0, i1)
loop domain: [ i3, i4, i5 ]
```
if `i3` is mapped to `i0` and `i5` is mapped to `i1`, then this will create a new id `i6 = merge(i3, i5)` and the new loop domain will be `[ i4, i6 ]` (and not `[ i6, i4 ]`. I think this makes sense but we should mention this in the comment in the header in my opinion.

Reply:
Actually, no, it should be `[i6, i4]` since the insertion of the new ID is done at `outermost_pos`. This should be the same behavior as `TensorView::merge`.

Yes, I'll add comments and also add a new test to verify the behavior.

Reply:
Test added